src/librustc_codegen_llvm/back/link.rs

```diff
@@ -609,6 +594,71 @@ fn print_native_static_libs(sess: &Session, all_native_libs: &[NativeLibrary]) {
     }
 }
 
+pub fn linker_and_flavor(sess: &Session) -> Result<(PathBuf, LinkerFlavor), ()> {
+    fn from<F>(
+        sess: &Session,
+        linker: Option<PathBuf>,
+        flavor: Option<LinkerFlavor>,
+        otherwise: F,
+    ) -> Result<(PathBuf, LinkerFlavor), ()>
+    where
+        F: FnOnce() -> Result<(PathBuf, LinkerFlavor), ()>
+    {
+        match (linker, flavor) {
+            (Some(linker), Some(flavor)) => Ok((linker, flavor)),
+            // only the linker flavor is known; use the default linker for the selected flavor
+            (None, Some(flavor)) => Ok((PathBuf::from(match flavor {
+                LinkerFlavor::Em => "emcc",
+                LinkerFlavor::Gcc => "gcc",
+                LinkerFlavor::Ld => "ld",
+                LinkerFlavor::Msvc => "link.exe",
```

Comment:
I think this is losing the logic of locating `link.exe` on MSVC which may cause issues there?

Reply:
The logic is in `get_linker` which is called with the values this function returns. I think that function will locate link.exe even with these changes.